src/pharmpy/model/statements.py

```diff
@@ -1416,6 +1557,41 @@ def amount(self):
 
     @property
     def dose(self):
+        if self._dose is not None:
+            return tuple(sorted(self._dose, key=lambda d: d.admid))
+        else:
+            return self._dose
+
+    @property
+    def oral_dose(self):
+        if self._dose is not None:
+            oral_dose = tuple([dose for dose in self.dose if dose.admid == 1])
+            return oral_dose if oral_dose else None
+        return None
+
+    @property
+    def iv_dose(self):
+        if self._dose is not None:
+            iv_dose = tuple([dose for dose in self.dose if dose.admid == 2])
+            return iv_dose if iv_dose else None
+        return None
+
+    @property
+    def number_of_doses(self):
```

Comment:
Is this property needed? Using `len()` is not so much longer?

Reply:
No, and has now been removed to avoid adding unnecessary properties